mmaction/localization/ssn_utils.py

```diff
@@ -0,0 +1,145 @@
+import fnmatch
+import glob
+import os
+from itertools import groupby
+
+
+def load_localize_proposal_file(filename):
+    lines = list(open(filename))
+
+    # Split the proposal file into many parts which contain one video's
+    # information separately.
+    groups = groupby(lines, lambda x: x.startswith('#'))
+
+    video_infos = [[x.strip() for x in list(g)] for k, g in groups if not k]
+
+    def parse_group(video_info):
+        """Template information of a video in a standard file:
+
+            # index
+            video_id
+            num_frames
+            fps
+            num_gts
+            label, start_frame, end_frame
+            label, start_frame, end_frame
+            ...
+            num_proposals
+            label, best_iou, overlap_self, start_frame, end_frame
+            label, best_iou, overlap_self, start_frame, end_frame
+            ...
+        Example of a standard annotation file:
+        .. code-block:: txt
+            # 0
+            video_validation_0000202
+            5666
+            1
+            3
+            8 130 185
+            8 832 1136
+            8 1303 1381
+            5
+            8 0.0620 0.0620 790 5671
+            8 0.1656 0.1656 790 2619
+            8 0.0833 0.0833 3945 5671
+            8 0.0960 0.0960 4173 5671
+            8 0.0614 0.0614 3327 5671
+        """
+        offset = 0
+        video_id = video_info[offset]
+        offset += 1
+
+        num_frames = int(float(video_info[1]) * float(video_info[2]))
+        num_gts = int(video_info[3])
+        offset = 4
+
+        gt_boxes = [x.split() for x in video_info[offset:offset + num_gts]]
+        offset += num_gts
+        num_proposals = int(video_info[offset])
+        offset += 1
+        proposal_boxes = [
+            x.split() for x in video_info[offset:offset + num_proposals]
+        ]
+
+        return video_id, num_frames, gt_boxes, proposal_boxes
+
+    return [parse_group(video_info) for video_info in video_infos]
+
+
+def process_localize_proposal_list(norm_proposal_list, out_list_name,
+                                   frame_dict):
+    norm_proposals = load_localize_proposal_file(norm_proposal_list)
+
+    processed_proposal_list = []
+    for idx, proposal in enumerate(norm_proposals):
+        video_id = proposal[0]
+        frame_info = frame_dict[video_id]
+        num_frames = frame_info[1]
+        frame_path = frame_info[0].split('/')[-1]
+
+        gts = [[
+            int(x[0]),
+            int(float(x[1]) * num_frames),
+            int(float(x[2]) * num_frames)
+        ] for x in proposal[2]]
+
+        proposals = [[
+            int(x[0]),
+            float(x[1]),
+            float(x[2]),
+            int(float(x[3]) * num_frames),
+            int(float(x[4]) * num_frames)
+        ] for x in proposal[3]]
```

Comment:
```
proposals = []
for x in proposal[3]:
    p = [int(x[0]), float(x[1]), float(x[2]), int(float(x[3]) * num_frames), int(float(x[4]) * num_frames)]
    proposals.append(p)
```
Also check whether the `float()` for `x[3]` and `x[4]` are necessary.

Reply:
Yes, we have to convert 'str' into 'float'.